tests/unit/workflows/test_workflows_tasks_submission.py

```diff
@@ -22,22 +22,196 @@
 
 from __future__ import absolute_import, division, print_function
 
-from inspirehep.modules.workflows.tasks.submission import add_note_entry
+from six import StringIO
+
+from inspirehep.modules.workflows.tasks.submission import (
+    add_note_entry,
+    prepare_files,
+)
+
+
+class MockLog(object):
+    def __init__(self):
+        self._debug = StringIO()
+        self._info = StringIO()
+
+    def debug(self, message):
+        self._debug.write(message)
+
+    def info(self, message):
+        self._info.write(message)
+
+
+class AttrDict(dict):
+    def __init__(self, *args, **kwargs):
+        super(AttrDict, self).__init__(*args, **kwargs)
+        self.__dict__ = self
+
+
+class StubFiles(object):
+    def __init__(self, data):
+        self.data = data
+
+    def __getitem__(self, key):
+        return self.data[key]
+
+    def __len__(self):
+        return len(self.data)
+
+    @property
+    def keys(self):
+        return self.data.keys()
 
 
 class StubObj(object):
-    def __init__(self, data, extra_data):
+    def __init__(self, data, extra_data, files):
         self.data = data
         self.extra_data = extra_data
+        self.files = files
+
+        self.log = MockLog()
 
 
 class DummyEng(object):
     pass
 
 
 def test_add_note_entry_does_not_add_value_that_is_already_present():
-    obj = StubObj({'public_notes': [{'value': '*Temporary entry*'}]}, {'core': 'something'})
+    data = {
+        'public_notes': [
+            {'value': '*Temporary entry*'},
+        ],
+    }
+    extra_data = {'core': 'something'}
+    files = StubFiles({})
+
+    obj = StubObj(data, extra_data, files)
     eng = DummyEng()
 
     assert add_note_entry(obj, eng) is None
-    assert {'public_notes': [{'value': '*Temporary entry*'}]} == obj.data
+    assert obj.data == {
+        'public_notes': [
+            {'value': '*Temporary entry*'},
+        ],
+    }
+
+
+def test_prepare_files():
+    data = {}
+    extra_data = {}
+    files = StubFiles({
+        'foo.pdf': AttrDict({
+            'obj': AttrDict({
+                'file': AttrDict({
+                    'uri': '/data/foo.pdf',
+                }),
+            }),
+        }),
+    })
+
+    obj = StubObj(data, extra_data, files)
+    eng = DummyEng()
+
+    assert prepare_files(obj, eng) is None
+    assert obj.data == {
+        'fft': [
+            {
+                'url': '/data/foo.pdf',
+                'docfile_type': 'INSPIRE-PUBLIC',
+                'filename': 'foo',
+                'filetype': '.pdf',
+            },
+        ],
+    }
+    assert 'Non-user PDF files added to FFT.' == obj.log._info.getvalue()
+    assert '/data/foo.pdf' in obj.log._debug.getvalue()
+
+
+def test_prepare_files_annotates_files_from_arxiv():
+    data = {
+        'arxiv_eprints': [
+            {'categories': 'hep-th'},
+        ],
+    }
```

Comment:
@rikirenz this is what I meant about the classicist vs. mockist: you don't need to mock everything.

Something it's fine if a test "leaks" responsibilities to the collaborators of the method by not mocking *everything* the method depends on. In my experience, code becomes more robust, because critical pieces are tested many times over in several different conditions, and bugs are not hidden by too aggressive mocks.